internal_workflow.go

```diff
@@ -327,6 +328,7 @@ func (d *syncWorkflowDefinition) Execute(env workflowEnvironment, input []byte)
 	d.rootCtx = WithValue(background, workflowEnvironmentContextKey, env)
 	var resultPtr *workflowResult
 	d.rootCtx = WithValue(d.rootCtx, workflowResultContextKey, &resultPtr)
+	d.rootCtx = WithValue(d.rootCtx, changeVersionsContextKey, make(map[string]Version))
```

Comment:
This layer server as a cache so we always return same version for same changeID, only if the changeID is known ID. The lookup at workflowEnvironment layer serve lookup from marker result. If a changeID is not present at workflowEnvironment layer, we would return DefaultVersion, but if a changeID is not present at here, we won't return DefaultVersion.